### PR TITLE
Move leading comment nodes into the document fragment

### LINKED ISSUE #9861: First comment node is missing after calling `DOMParser.parseFromString()`
## Provide a description of the task

This issue has been found during working on HTML comments.

The DOM string parsed by [`HtmlDataProcessor#_toDom()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_dataprocessor_htmldataprocessor-HtmlDataProcessor.html#function-_toDom) does not contain the first comment node, if it is the first node in the DOM string.

This issue is caused because the return value from:
```js
new DOMParser().parseFromString( '<!--COMMENT 1--><p>PARAGRAPH 1</p><!--COMMENT 2-->', 'text/html' ).body.childNodes
```
is a two-elements list: `NodeList [ p, <!--COMMENT 2--> ]` without the first comment node `<!--COMMENT 1-->`. The first comment is somehow hoisted and inserted before the newly created HTML document:

This issue could be fixed by wrapping the DOM string with a `<body>` or even better with a `<div>`, just like it is done for [`BasicHtmlWriter#getHtml()`](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_dataprocessor_basichtmlwriter-BasicHtmlWriter.html#function-getHtml)

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Other (engine): Fixed parsing leading HTML comments by `HtmlDataProcessor.toView()`. Closes #9861.

### CODE REVIEW

File: packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.js [hunk 1]
```diff
@@ -119,10 +119,36 @@ export default class HtmlDataProcessor {
...
+
+		while ( !documentChildNode.isSameNode( document.documentElement ) ) {
+			const node = documentChildNode;
+
+			documentChildNode = documentChildNode.nextSibling;
+
+			/* istanbul ignore else */
+			if ( node.nodeType == Node.COMMENT_NODE ) {
+				fragment.appendChild( node );
+			}
```
Comment (psmyrek) [inline packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.js:145]: Probably only comment nodes can be added directly to the Document object, before the `<html>` node. I have not been able to create any HTML string that would insert a node other than a comment at the beginning of the Document object, so the `/* istanbul ignore else */` is here to keep the 100% CC.

Anyway, this `if` condition here is just to be sure.

Comment (ma2ciek) [reply packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.js:145]: Maybe it could be moved to the comment.

Comment (psmyrek) [reply packages/ckeditor5-engine/src/dataprocessor/htmldataprocessor.js:148]: Comment has been added.

I also replaced the `Document` with `HTMLDocument` in comments, as the latter name [is used in the specification](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString#return_value).

**Review by ma2ciek — APPROVED** [review]
LGTM. @niegowski, maybe you could take a look as well.

**Review by niegowski — APPROVED** [review]
👍  LGTM

**Comment by niegowski** [comment]
But the merge commit message should indicate that it fixes parsing the leading HTML comments.

**Comment by psmyrek** [comment]
Commit message is updated. Is it better now? 🙂

**Comment by niegowski** [comment]
For the commit message, I think there are too many internal details. I'd go just with:

Fixed parsing leading HTML comments with `HtmlDataProcessor.toView()`.

**Pull request merged**